react/CheckPermission.tsx

```diff
@@ -9,27 +9,28 @@ interface Props {
   roles: string[]
   AllowedContent?: ElementType
   DisallowedContent?: ElementType
+  LoadingContent?: ElementType
 }
 
 function CheckPermission({
   roles = [],
   AllowedContent,
   DisallowedContent,
+  LoadingContent,
 }: Props) {
   const { data, called, error, loading } = useQuery(getPermissions, {
     ssr: false,
     skip: !roles.length,
   })
 
-  if (!roles.length || !data) {
-    return null
+  if (!roles.length || !data || (called && loading)) {
+    return LoadingContent
+      ? <LoadingContent />
+      : <ExtensionPoint id="loading-content" />
   }
 
-  if ((called && loading) || error) {
-    if (error !== undefined) {
-      console.error('CheckPermission error:', error)
-    }
-
+  if (error !== undefined) {
```

Comment:
I believe that this block needs to be before the loading check on line 26, otherwise the `!data` part of the check will  make it always show a loading spinner for error cases. Handling the error first, if any exists, is safer.

Reply:
Also, I believe it's better to just check for `if (error)`, as `null !== undefined` is true. 

```suggestion
  if (error) {
```

Reply:
Suggestions accepted and pushed @mairatma